### PR TITLE
test(simple-color-picker): refactor cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor Cypress tests for the `Simple-Color-Picker` component to use new cypress-component-react framework for testing.
Add `simple-color-picker.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `simple-color-picker.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/simple-color-picker/simple-color-picker.test.js [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+          "have.attr",
+          "aria-checked",
+          "true"
+        );
+      }
+    );
+
+    it.each([
+      ["250", "232px"],
+      ["450", "406px"],
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply src/components/simple-color-picker/simple-color-picker.test.js:215]: it accepts only `string`'s :)

**Review by stephenogorman — APPROVED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

File: src/components/simple-color-picker/simple-color-picker.test.js [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+
+      simpleColorPickerInput(5)
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it.each([
```
Comment (edleeks87) [inline src/components/simple-color-picker/simple-color-picker.test.js:344]: I think we can write these tests to assert that a relevant `SimpleColor` square is focussed and that the callback stub is called with the relevant colour. We should do this for `right`/ `left` `up` and `down` imo

```
const indexes = Array.from({ length: colors.length }).map((_, index) => index);

it("should call onChange callback and focus the correct item when the right arrow key is triggered", () => {
    CypressMountWithProviders(
          <SimpleColorPickerCustom onChange={callback} />
        );
        simpleColorPickerInput(0).click()

        indexes.forEach((index) => {
          const next = index < colors.length - 1 ? index + 1 : 0;

          simpleColorPickerInput(index)
          .trigger("keydown", keyCode("rightarrow"))
          .then(() => {
            // eslint-disable-next-line no-unused-expressions
            expect(callback).to.have.been.calledWith(colors[next].color);
          })
          .then(() => {
              simpleColorPickerInput(next).should("be.focused")
            });
        })
      }
    );

    it("should call onChange callback and focus the correct item when the left arrow key is triggered", () => {
      CypressMountWithProviders(
            <SimpleColorPickerCustom onChange={callback} />
          );
          simpleColorPickerInput(0).click()
  
          indexes.reverse().forEach((index) => {
            const next = index > 0 ? index - 1 : colors.length - 1;
  
            simpleColorPickerInput(index)
            .trigger("keydown", keyCode("leftarrow"))
            .then(() => {
              // eslint-disable-next-line no-unused-expressions
              expect(callback).to.have.been.calledWith(colors[next].color);
            })
            .then(() => {
                simpleColorPickerInput(next).should("be.focused")
              });
          })
        }
      );
  ```

Comment (DlgSHi) [reply src/components/simple-color-picker/simple-color-picker.test.js:388]: thx :) applied

**Review by stephenogorman — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**